### PR TITLE
Fix: 14814 --- Inaccurate error in anonymous record copy-and-update

### PR DESCRIPTION
**Pull request by KevinRansom**
This error was introduced with this commit:  https://github.com/dotnet/fsharp/commit/2ef33c4edf540805d78e591b083d7da9396c996a

Checked in code:
image

The fix is to revert the breaking commit:
image

Someone should revisit the original commit to see if it can be made robust to the repro in 14814

### CODE REVIEW

**Comment by vzarytovskii** [comment]
This ended up in 17.7 and 7.0, will need a backport to 17.7 and tactics...sigh

**Review by T-Gro — APPROVED** [review]

File: tests/service/Symbols.fs [hunk 1]
```diff
@@ -474,207 +474,3 @@ type Foo =
           (:? FSharpMemberOrFunctionOrValue as setMfv) ->
             Assert.AreNotEqual(getMfv.CurriedParameterGroups, setMfv.CurriedParameterGroups)
         | _ -> Assert.Fail "Expected symbols to be FSharpMemberOrFunctionOrValue"
-        
-    [<Test>]
```
Comment (nojaf) [inline tests/service/Symbols.fs:478]: @KevinRansom why did you remove these tests?
You really removed a bunch of these that are not even related!

Comment (vzarytovskii) [reply tests/service/Symbols.fs:478]: @nojaf it was just plain revert of the following commit:
https://github.com/dotnet/fsharp/commit/2ef33c4edf540805d78e591b083d7da9396c996a

Comment (vzarytovskii) [reply tests/service/Symbols.fs:478]: Or maybe not?

Comment (nojaf) [reply tests/service/Symbols.fs:478]: Clearly not: https://github.com/dotnet/fsharp/commit/2ef33c4edf540805d78e591b083d7da9396c996a#diff-86de212e8dc5b638058239f2b2b485cd6207f61b3acb2556de8bb7fef15065ea

Tests are removed that were not introduced by https://github.com/dotnet/fsharp/pull/15214

Comment (vzarytovskii) [reply tests/service/Symbols.fs:478]: Seems mistakenly removed. I will re-add them

Comment (nojaf) [reply tests/service/Symbols.fs:478]: Thanks

**Comment by auduchinok** [comment]
> @auduchinok fyi

Thanks. I've just checked, and it doesn't seem to be connected to https://github.com/dotnet/fsharp/pull/15214 (more details in https://github.com/dotnet/fsharp/issues/14814#issuecomment-1642146536)
